blueprints/token.lua

```diff
@@ -129,11 +129,146 @@ end)
    ]]
 --
 Handlers.add('balances', Handlers.utils.hasMatchingTag("Action", "Balances"),
-  function(msg) 
+  function(msg)
     if msg.reply then
       msg.reply({ Data = json.encode(Balances) })
-    else 
-      Send({Target = msg.From, Data = json.encode(Balances) }) 
+    else
+      Send({Target = msg.From, Data = json.encode(Balances) })
+    end
+  end)
+
+   --[[
+     Batch-Transfer
+     Processes multiple transfers atomically from a CSV input
+   ]]
+ --
+Handlers.add('batch-transfer', Handlers.utils.hasMatchingTag("Action", "Batch-Transfer"), function(msg)
```

Comment:
Recommend adding a max message size and throwing if it exceeds it. Each transfer is going to create a credit notice, so for example if you do a 100k size transfer its going to create 100k notices, which gets JSON encoded, and that consumes a ton of memory. I recommend limiting this to something like 1k entries per batch transfer.

Reply:
The motivation behind this implementation was specifically to avoid sending thousands of debit and credit notices; otherwise, you could simply run transfers in a loop. The recommended behavior is therefore to set the `CAST` tag and completely avoid sending credit notices. For consistency with the transfer implementation, there is an option to send credit notices, but this is again not recommended for large batches.